migrate/v0_15/account_test.go

```diff
@@ -0,0 +1,284 @@
+	assert.Equal(t, spendableBefore, vacc.SpendableCoins(newVestingStartTime), "expected spendable coins to be unchanged")
+}
+
+func TestResetPeriodVestingAccount_SingleVestingPeriod_ExactStartTime(t *testing.T) {
+	balance := sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(1e6)))
+	vestingStartTime := time.Now().Add(-30 * 24 * time.Hour) // 30 days in past
+
+	periods := vesting.Periods{
+		vesting.Period{
+			Length: 30 * 24 * 60 * 60, // 30 days - exact on the start time
+			Amount: balance,
+		},
+	}
+
+	vacc := createVestingAccount(balance, vestingStartTime, periods)
+
+	newVestingStartTime := vestingStartTime.Add(30 * 24 * time.Hour)
+	spendableBefore := vacc.SpendableCoins(newVestingStartTime)
+
+	ResetPeriodicVestingAccount(vacc, newVestingStartTime)
+
+	// new period length is 0
+	expectedEndtime := newVestingStartTime.Unix()
+	// new period length changed, amount unchanged
+	expectedPeriods := vesting.Periods{}
+
+	assert.Equal(t, sdk.Coins{}, vacc.OriginalVesting, "expected original vesting to be unchanged")
+	assert.Equal(t, newVestingStartTime.Unix(), vacc.StartTime, "expected vesting start time to be updated")
+	assert.Equal(t, expectedEndtime, vacc.EndTime, "expected vesting end time end at last period")
+	assert.Equal(t, expectedPeriods, vacc.VestingPeriods, "expected vesting periods to be updated")
+	assert.Equal(t, spendableBefore, vacc.SpendableCoins(newVestingStartTime), "expected spendable coins to be unchanged")
+}
+
+func TestResetPeriodVestingAccount_MultiplePeriods(t *testing.T) {
+	balance := sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(3e6)))
+	vestingStartTime := time.Now().Add(-30 * 24 * time.Hour) // 30 days in past
+
+	periods := vesting.Periods{
+		vesting.Period{
+			Length: 15 * 24 * 60 * 60, // -15 days - vested
+			Amount: sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(1e6))),
+		},
+		vesting.Period{
+			Length: 30 * 24 * 60 * 60, // 0 days - exact on the start time
+			Amount: sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(1e6))),
+		},
+		vesting.Period{
+			Length: 45 * 24 * 60 * 60, // +15 days - vesting
+			Amount: sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(1e6))),
+		},
+	}
+
+	vacc := createVestingAccount(balance, vestingStartTime, periods)
+
+	newVestingStartTime := vestingStartTime.Add(30 * 24 * time.Hour)
+	spendableBefore := vacc.SpendableCoins(newVestingStartTime)
+
+	ResetPeriodicVestingAccount(vacc, newVestingStartTime)
+
+	// new period length 15 days
+	expectedEndtime := newVestingStartTime.Add(15 * 24 * time.Hour).Unix()
+	// new period length changed, amount unchanged
+	expectedPeriods := vesting.Periods{
+		vesting.Period{
+			Length: 0, // 0 days
+			Amount: sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(1e6))),
+		},
+		vesting.Period{
+			Length: 15 * 24 * 60 * 60, // 15 days
+			Amount: sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(1e6))),
+		},
+	}
+
+	assert.Equal(t, sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(2e6))), vacc.OriginalVesting, "expected original vesting to be unchanged")
```

Comment:
"expected original vesting to be changed". correct? Looks like it goes from 3e6 to 2e6